src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('.employees-table');
+
+  if (!table) {
+    return;
+  }
+
+  const headers = table.querySelectorAll('th');
+  const rows = Array.from(table.querySelectorAll('tbody tr'));
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      const isNumeric = header.dataset.type === 'number';
+      const sortedRows = rows.sort((rowA, rowB) => {
+        const cellA = rowA.children[index].textContent.trim();
+        const cellB = rowB.children[index].textContent.trim();
+
+        if (isNumeric) {
+          return parseFloat(cellA) - parseFloat(cellB);
```

Comment:
Consider adding a check to handle cases where `parseFloat` might return `NaN` if the cell content is not a valid number. This could prevent unexpected sorting behavior.